### PR TITLE
Utilize the k8s-reporter in the e2e tests

### PR DESCRIPTION
**Pull request by liornoy**
Modify the k8sreporter/reporter.go to utilize
the latest external k8s-reporter package.

The new reporter still outputs the same output as before plus the CRDs: SriovOperatorConfig and SriovNetwork.

It is used by the conformance and validation suites, consumes the flag "-report" as the path to where the dir "sriov_failure_report.log" will be created and hold all of the failed tests logs.

Here is an example of the k8s reporter tree output when all the tests of the validation suite failed:
```
.
└── sriov_failure_report.log
    ├── validation_sriov_should_deploy_the_injector_pod_if_requested
    │   ├── crs.log
    │   └── nodes.log
    ├── validation_sriov_should_deploy_the_operator_webhook_if_requested
    │   ├── crs.log
    │   └── nodes.log
    ├── validation_sriov_should_have_SR-IOV_node_statuses_not_in_progress
    │   ├── crs.log
    │   └── nodes.log
    ├── validation_sriov_Should_have_the_sriov_CRDs_available_in_the_cluster
    │   ├── crs.log
    │   └── nodes.log
    ├── validation_sriov_should_have_the_sriov_namespace
    │   ├── crs.log
    │   └── nodes.log
    └── validation_sriov_should_have_the_sriov_operator_deployment_in_running_state
        ├── crs.log
        └── nodes.log
```
Where the crs.log will dump all of the CRDs,
nodes.log is the list of nodes objects in json format,
and in case pods exist in the sriov namespace, it will dump
for each test: pods_logs.log and pods_specs.log

For more details about the reporter see here: https://github.com/openshift-kni/k8sreporter

### CODE REVIEW

**Review by zeeke — APPROVED** [review]
LGTM

**Comment by SchSeba** [comment]
@Eoghan1232 can you please take a look as we need another company maintainer to merge this PR

File: test/conformance/test_suite_test.go [hunk 1]
```diff
@@ -2,81 +2,55 @@ package conformance
...
-
-	clients := testclient.New("")
+	// We want to collect logs before any resource is deleted in AfterEach, so we register the global fail handler
+	// in a way such that the reporter's Dump is always called before the default Fail.
+	RegisterFailHandler(
+		func(message string, callerSkip ...int) {
+			if customReporter != nil {
+				customReporter.Dump(10*time.Minute, CurrentSpecReport().FullText())
+			}
+			Fail(message, callerSkip...)
```
Comment (zeeke) [inline test/conformance/test_suite_test.go:40]: Please consider increasing `callerSkip` slice, as described here:

- https://github.com/openshift-kni/cnf-features-deploy/pull/1644

```suggestion
                        // Ensure failing line location is not affected by this wrapper
			for i := range callerSkip {
				callerSkip[i]++
			}
			Fail(message, callerSkip...)
```

Comment (liornoy) [reply test/conformance/test_suite_test.go:45]: Done. thanks!

**Review by Eoghan1232 — APPROVED** [review]
lgtm! much cleaner way to do it!

**Pull request merged**